glib/TypeEqualityComparer.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace GLib
+{
+	class TypeEqualityComparer : IEqualityComparer<Type>
+	{
+		public bool Equals (Type x, Type y)
```

Comment:
It looks like we should be able to rely on the default comparer here and we don't gain anything from the custom implementation. Why can't we just remove this completely?

Reply:
There is a perf implication. Xamarin.Mac uses this too

Reply:
Huh, odd. It's a reference type and the custom implementation is just calling  the built in equality methods. I won't argue the point, I just find it quite weird :p

Reply:
I wonder if that was for AOT for xammac.

Reply:
I think this was to workaround Type equality being slow in older versions of Mono.

Reply:
Not older versions, seems like .netframework RuntimeType still doesn't overload object.Equals